lib/exprengine.cpp

```diff
@@ -971,6 +975,32 @@ static bool isNonOverlapping(ExprEngine::ValuePtr v1, ExprEngine::ValuePtr v2)
     return false;
 }
 
+static bool conditionAlwaysFalse(ExprEngine::ValuePtr condValue, ExprEngine::DataBase *dataBase)
+{
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue)) {
+        return v->hasValue(dataBase, 0);
```

Comment:
I doubt the `hasValue` would work well because `x` will be an IntRange with the value 0-255 in these test cases.

I have merged these test cases and a small fix with 5701f6d3687740f53ad06e27fd9dd883ad6d08a9